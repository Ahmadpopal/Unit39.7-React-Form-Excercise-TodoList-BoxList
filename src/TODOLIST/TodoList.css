.TodoList-title {
  text-align: center;
}

.TodoList-subtitle {
  text-align: center;
  margin: 0;
}
