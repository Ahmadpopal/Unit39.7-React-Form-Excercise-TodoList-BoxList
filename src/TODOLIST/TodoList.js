import React, {useState} from 'react';
import NewTodoForm from './NewTodoForm';
import {v4 as uuid} from 'uuid'
import Todo from "./Todo"
import './TodoList.css'




const TodoList= () => {
    const INITIAL_STATE = [
        // {id: 1, todo: "go to work"}, 
        // {id: 2, todo: "Attend the Meeting"}, 
        // {id: 3, todo: "Leave At 4 pm"}
    ]


    const [todos, setTodos] = useState(INITIAL_STATE)


    function addTodo(todoObj){
        setTodos((todo) => [...todos, {id: uuid(), ...todoObj}])
    }

    function removeTodo(id){
        setTodos(todos => todos.filter(todo => todo.id !== id));
    }



    return (
        <div className="TodoList">
            <h1 className="TodoList-title">Todo List</h1>
            <NewTodoForm  addTodo={addTodo}/>
            <h2 className="TodoList-subtitle">Todo List</h2>
            {todos.map((todos) => (
                <Todo key={todos.id} todo={todos.todo} id={todos.id} removeTodo={removeTodo}/>
            ))}
        </div>
    )
}

export default TodoList