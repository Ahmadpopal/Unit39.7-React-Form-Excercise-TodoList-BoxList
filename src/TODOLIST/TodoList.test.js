import React from "react";
import {render, fireEvent, getByLabelText } from '@testing-library/react';
import TodoList from "./TodoList"


// SMOKE TEST 
it('render without crashing', () => {
    render(<TodoList />);
});

// SNAPSHOT TEST 
it('it matches snapshot', () =>{
    const {asFragment } = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
})