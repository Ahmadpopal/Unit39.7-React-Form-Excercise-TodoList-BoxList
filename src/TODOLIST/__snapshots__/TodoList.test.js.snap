// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="TodoList"
  >
    <h1
      class="TodoList-title"
    >
      Todo List
    </h1>
    <form
      class="NewTodoForm"
    >
      <label
        class="NewTodoForm-label"
        for="todo"
      >
        New Todo
      </label>
      <input
        class="NewTodoForm-input"
        id="tode"
        name="todo"
        placeholder="Add Todos"
        type="text"
        value=""
      />
      <button
        class="NewTodoForm-btn"
      >
        Add Todo
      </button>
    </form>
    <h2
      class="TodoList-subtitle"
    >
      Todo List
    </h2>
  </div>
</DocumentFragment>
`;
